### PR TITLE
Initial editorconfig setup

### PR DESCRIPTION
**Pull request by oguzkocer**
I went back and forth on what we would want in the initial `editorconfig` setup and ended up with this PR.

```
[*]
end_of_line = lf
insert_final_newline = true
trim_trailing_whitespace=true
```

I think we want these 3 rules for all files and they might be enough for what we need out of the `editorconfig` file. Specifically, `trim_trailing_whitespace` is the one causing issue for us since some editors are setup to do that and others are not, so it creates commits with extra whitespace changes. The `end_of_line` is just for the peace of mind and `trim_trailing_whitespace` is a good default.

```
[*.rb]
indent_style = space

[**FastFile]
indent_style = space
```

This is where I spent most of my time. I thought we would also add the `indent_size` as 2 [as per a seemingly popular style guide](https://github.com/rubocop-hq/ruby-style-guide#indentation). However, if I try to fix our current ruby files with that option, function arguments go out of alignment, [which partially happened in this PR even without that rule](https://github.com/wordpress-mobile/WordPress-Android/compare/add/editorconfig?expand=1#diff-e70f81b97fbe6296a7ae8fe4085db78bL48-R74). I am not too sure what to do about it.

I also didn't want to auto-fix every file in the repository. The project itself has its own rules, so it doesn't really need the `.editorconfig` rules and most other files are not altered that often. So, I just ran the auto-fix for the files under `fastlane` directory which is probably enough for us. We can fix any other issues as we touch the files and with [Hide whitespace changes option in Github PRs](https://cloudup.com/c9eOMwgSPYL) it shouldn't be too hard for us to ignore them.

Finally, for the auto-fix, I used [eclint](https://www.npmjs.com/package/eclint) which is an npm tool that uses `.editorconfig` file to check/fix styling issues.

---

**I'd like us to use this PR:**
1. To settle on a set of rules
2. To decide which files we want to auto-fix right now (should we for example remove the changes for txt/html files)
3. To figure out if we want to automate anything (by running a tool such as eclint)

_P.S: I'll share my thoughts as a comment as to not clutter the PR description._

**PR submission checklist:**

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by oguzkocer** [comment]
> To settle on a set of rules

I like the ruleset from this PR as a starting point. It might be nice to add the `indent_space` for ruby files, but I think we should consider adding proper linting for Ruby instead of relying on `.editorconfig`. A lot of our team's work is with Ruby, so it makes sense to me to spend a bit of time to set it all up. If it's at all possible, I'd love to have a remote config file which we can use in all our projects.

> To decide which files we want to auto-fix right now (should we for example remove the changes for txt/html files)

I'd just keep the changes from this PR just because it's already done. However, I wouldn't worry about the files we generate on an on-going basis. If we have some extra whitespace there, it won't really matter.

> To figure out if we want to automate anything (by running a tool such as eclint)

It's probably not worth doing this. If all 3 of us have our editors setup to use the `.editorconfig` file, there won't be a major need for it. If a whitespace creeps in a PR, so be it 🤷 We just want to prevent big changesets in PRs due to these whitespace issues and having our editors setup is good enough for me.

Having said that, if we do implement proper ruby linting, that would be worth automating for me. For one is proper style guide that improves coding and the other is just a minor annoyance.

---

Looking forward to hear your thoughts as well!

**Comment by loremattei** [comment]
Thanks @oguzkocer! 
I like this. 

>  I thought we would also add the indent_size as 2

Actually, I'd add this as well. If auto-fixing doesn't work, I'm fine with fixing it when we happen to work on a file, but I'd add the rule (if there aren't side effects that I don't figure out now).

I agree with your other thoughts in your comment above.

**Comment by loremattei** [comment]
I'm approving this, but not merging so that we can gather feedback from Gio and Jeremy as well.

**Review by loremattei — APPROVED** [review]

File: .editorconfig [hunk 1]
```diff
@@ -1,2 +1,13 @@
...
+insert_final_newline = true
+trim_trailing_whitespace=true
+
 [*.{kt,kts}]
 max_line_length=120
+
+[*.rb]
+indent_style = space
+
+[**FastFile]
```
Comment (mokagio) [inline .editorconfig:12]:
```suggestion
[**Fastfile]
```

From the [docs](https://editorconfig.org/#file-format-details), these patters are case sensitive.

Screen Shot 2020-07-13 at 4 01 59 pm


I tried to add `indent_size = 16` here, and it only worked with `[**Fastfile]`, not `[**Fastfile]`.

**Review by mokagio — COMMENTED** [review]
Thanks @oguzkocer for getting the ball rolling on this 👍 

> I'd like us to use this PR:
>
> - To settle on a set of rules
> - To decide which files we want to auto-fix right now (should we for example remove the changes for txt/html files)
> - To figure out if we want to automate anything (by running a tool such as eclint)

I think these are a good starting point, almost a matter of text file hygiene, regardless of the programming language used. I think adopting the defaults suggested by Rubocop like you did makes sense, too. It is the default styleguide I've seen used in Ruby so far.

>  I just ran the auto-fix for the files under fastlane directory which is probably enough for us. We can fix any other issues as we touch the files.

I think this makes sense, too. It's a logical starting point. Another option I'd be fine with would be autofixing _everything_, in the spirit of taking the plunge and not having to think about it anymore.

Finally, regarding automation, I think integrating eclint or similar in the CI process is a good idea. Under the assumption that most contributors to the codebase will have EditorConfig integrated in their preferred setup (something we might want to suggest in the README?) this kind of warnings/errors in CI should occur rarely.

---

I'm not approving this just because of the `FastFile` vs `Fastfile` comment, otherwise, I think it's ready to go. It'd be curious to know if your setup is case insensitive and/or if changing the pattern helps with adding support for the indentation size rule, too.

File: .editorconfig (hunk 1 above)
Comment (jkmassel) [reply .editorconfig:12]: Only having the first letter capitalized seems to be the norm, so I'm good to follow this here too :)

**Comment by jkmassel** [comment]
I think this is great – I'm definitely onboard to run it on whichever files it should be applied to as part of each commit in order to detect issues. If I'm doing a quick change, I might do it in terminal or something where my editor preferences may not be applied. To avoid having to fix it up later, it'd be nice to ensure we're always committing valid changes IMHO.

**Comment by oguzkocer** [comment]
I know this is a super simple PR, but I just can't find the time to work on it. I'd like to spend what little time I've left on the modularization project, so I am closing this PR. If any of you would like to take over, please feel free to take over the branch and re-open the branch. I am sorry!

**Comment by mokagio** [comment]
No stress!

**Pull request closed without merging**